lib/ExpensiMark.js

```diff
@@ -285,4 +292,87 @@ export default class ExpensiMark {
+
+            // split the textToCheck in lines
+            const textSplitted = textToCheck.split('\n');
+
+            for (let i = 0; i < textSplitted.length; i++) {
+                if (!insideCodefence) {
+                    // we need to know when there is a start of codefence so we dont quote
+                    insideCodefence = Str.contains(textSplitted[i], '<pre>');
+                }
+
+                // we only want to modify lines starting with &gt; that is not codefence
+                if (Str.startsWith(textSplitted[i], '&gt;') && !insideCodefence) {
+                    // avoid blank lines starting with &gt;
+                    if (textSplitted[i].trim() !== '&gt;') {
+                        textSplitted[i] = textSplitted[i].substr(4).trim();
+                        textSplitted[i] = textSplitted[i].trim();
+                        textToFormat += `${textSplitted[i]}\n`;
+
+                        // we need ensure that index i-1 >= 0
+                    } else if (i > 0) {
+                        // certificate that we will have only one blank row
+                        if (textSplitted[i - 1].trim() !== '') {
+                            textSplitted[i] = textSplitted[i].substr(4).trim();
+                            textSplitted[i] = textSplitted[i].trim();
+                            textToFormat += `${textSplitted[i]}\n`;
+                        }
+                    }
+                } else {
+                    // make sure we will only modify if we have Text needed to be formatted for quote
+                    if (textToFormat !== '') {
+                        textToFormat = textToFormat.replace(/\n$/, '');
+                        replacedText += replacement(textToFormat);
+                        textToFormat = '';
+                    }
+
+                    // if index === last row
+                    if (i === textSplitted.length - 1) {
+                        // we dont want a \n after the textSplitted if it is the last row
+                        replacedText += `${textSplitted[i]}`;
+                    } else {
+                        replacedText += `${textSplitted[i]}\n`;
+                    }
+
+                    // we need to know when we are not inside codefence anymore
+                    if (insideCodefence) {
+                        insideCodefence = !Str.contains(textSplitted[i], '</pre>');
+                    }
+                }
+            }
+
+            // when loop ends we need the last quote to be formatted if we have quotes at last rows
+            if (textToFormat !== '') {
+                textToFormat = textToFormat.replace(/\n$/, '');
+                replacedText += replacement(textToFormat);
+            }
+
+            // replace all the blank lines between quotes, if there are only blank lines between them
+            replacedText = replacedText.replace(/(<\/blockquote>((\s*)+)<blockquote>)/g, '</blockquote><blockquote>');
```

Comment:
I think you could move this outside the loop and just do this replacement once before returning from the function

Reply:
Its already outside the loop